RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/src/stc/PlatWX.cpp,v
retrieving revision 1.41
diff -u -r1.41 PlatWX.cpp
--- src/stc/PlatWX.cpp	2003/06/06 21:32:47	1.41
+++ src/stc/PlatWX.cpp	2003/07/21 23:46:14
@@ -481,7 +481,7 @@
     // so figure it out and fix it!
     i = 0;
     size_t ui = 0;
-    while (i < len) {
+    while ((int)i < len) {
         unsigned char uch = (unsigned char)s[i];
         positions[i++] = tpos[ui];
         if (uch >= 0x80) {
@@ -1215,6 +1215,22 @@
 
 
 //----------------------------------------------------------------------
+
+#if wxUSE_UNICODE
+wxString stc2wx(const char* str, size_t len)
+{
+    char *buffer=new char[len+1];
+    strncpy(buffer, str, len);
+    buffer[len]=0;
+
+    wxString cstr(buffer, wxConvUTF8);
+
+    delete[] buffer;
+    return cstr;
+}
+#endif
+
+
 
 
 
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/stc/stc.h,v
retrieving revision 1.37
diff -u -r1.37 stc.h
--- include/wx/stc/stc.h	2003/05/29 23:50:14	1.37
+++ include/wx/stc/stc.h	2003/07/21 23:46:19
@@ -2431,13 +2431,15 @@
 #endif
 }
 
-inline wxString stc2wx(const char* str, size_t len) {
+
 #if wxUSE_UNICODE
-    return wxString(str, wxConvUTF8, len);
+wxString stc2wx(const char* str, size_t len);
 #else
+inline wxString stc2wx(const char* str, size_t len) {
     return wxString(str, len);
-#endif
 }
+#endif
+
 
 #if wxUSE_UNICODE
 inline const wxWX2MBbuf wx2stc(const wxString& str) {

 	  	 
